components/applications-service/pkg/server/server.go

```diff
@@ -335,6 +335,29 @@ func (app *ApplicationsServer) DeleteDisconnectedServices(ctx context.Context,
 	}, nil
 }
 
+func (app *ApplicationsServer) DeleteServicesByID(ctx context.Context,
+	request *applications.DeleteServicesByIDReq) (*applications.ServicesRes, error) {
+	svcIDs := request.GetIds()
+	if svcIDs == nil {
+		return new(applications.ServicesRes),
+			status.Error(codes.InvalidArgument, "Must specify service IDs to delete")
+	}
+	if len(svcIDs) == 0 {
```

Comment:
i'm curious why the nil check == return error but the 0 id check returns empty?

Reply:
I had the same question, I remember defaulting to return `new()` for consistency.

Reply:
@afiune I think the question is why I chose to return `status.Error(codes.InvalidArgument, ....` when `request.GetIds()` is nil, but return an empty response for an empty request. The answer to that is that the nil case feels more like a malformed/incompletely formed request, whereas an empty request is kinda silly but correctly formed. You could also imagine that a user writes some periodic cleanup script and doesn't think to skip the delete request when there's zero services to cleanup; failing the request might trigger some alerts or something for little reason.

I'll change the ` &applications.ServicesRes{}` to `new` so it matches.

Reply:
that makes sense, thanks for explaining!

Reply:
that makes sense, thanks for explaining!